source/_docs/tools/dev-tools.markdown

```diff
@@ -24,3 +24,79 @@ Screenshot of Home Assistant's Developer Tools.
+
+{% linkable_title Services %}
+
+This section is used to call Services that are available in the ServiceRegistry.
+
+The list of services in the “Service” drop down are automatically populated based on the components that are found in the configuration, automation and script files.  If a desired service does not exist, it means either the component is not configured properly or not defined in the configuration, automation or script files.
+
+When a Service is selected, and if that service requires an `entity_id` to be passed, the “Entity” drop down will automatically be populated with corresponding entities.
+
+A Service may also require additional input to be passed. It is commonly referred to as “service data”. The service data is only accepted in the JSON format, and it may be optional depending on the service.
+
+When an entity is selected from the Entity drop down, it automatically populates service data with the corresponding `entity_id`. The service data JSON can then be modified to pass additional \[optional\] parameters. The following is an illustration on how to call a `light.turn_on` service.
+
+To turn on a light bulb, use the following steps:
+1.	Select `light.turn_on` from the Service drop down
+2.	Select the entity (typically the light bulb) from the Entity drop down (if no entity_id is selected, it turns on ALL lights)
+3.	If an entity is selected, the service data is populated with basic JSON that will be passed to the service. An additional data can also be passed by updating the JSON as below.
+
+```json
+{
+  "entity_id": "light.bedroom",
+  "brightness": 255,
+  "rgb_color": [255, 0, 0]
+}
+```
+{% linkable_title States %}
+
+This section shows all the available entities, their corresponding state and the attribute values. The state and the attribute information is what Home Assistant sees at run time. To update the entity with a new state, or a new attribute value, click on the entity, scroll to the top, and modify the values, and click on “SET STATE” button.
+
+Note that this is the state representation of a device within Home Assistant. That means, it is what Home Assistant sees, and it does not communicate with the actual device in any manner. The updated information can still be used to trigger events, and state changes. To communicate with the actual device, it is recommended to call services in the services section above, instead of updating state.
+
+For ex: Changing the `light.bedroom` state from `off` to `on` does not turn on the light. If there is an automation that triggers on the `state` change of the `light.bedroom`, it will be triggered – even though the actual bulb has not turned on. Also, when the bulb state changes – the state information will be overridden. In other words, the changes that are made through the “States” section are temporary, and is recommended to use for testing purposes only.
```

Comment:
For example, changing...

Reply:
last line
`...and is recommended for testing purposes only.`

Reply:
I think we are sending too much time going back and forth. It is crazy! Grammatically, it was correct.